wordclock/http_server.c

```diff
@@ -0,0 +1,1216 @@
+            *path_end = 0;
+
+            /* Dispatch to separate functions to handle the requests. */
+            const wificfg_dispatch *match = NULL;
+            const wificfg_dispatch *dispatch;
+
+            /*
+             * Check the optional application supplied dispatch table
+             * first to allow overriding the wifi config paths.
+             */
+            if (params->dispatch) {
+                for (dispatch = params->dispatch; dispatch->path != NULL; dispatch++) {
+                    if (strcmp(path_string, dispatch->path) == 0 &&
+                        (dispatch->method == HTTP_METHOD_ANY ||
+                         method == dispatch->method)) {
+                        match = dispatch;
+                        break;
+                    }
+                }
+            }
+
+            if (!match) {
+                for (dispatch = params->wificfg_dispatch; dispatch->path != NULL; dispatch++) {
+                    if (strcmp(path_string, dispatch->path) == 0 &&
+                        (dispatch->method == HTTP_METHOD_ANY ||
+                         method == dispatch->method)) {
+                        match = dispatch;
+                        break;
+                    }
+                }
+            }
+
+
+            /* Read the headers, noting some of interest. */
+            long content_length = 0;
+            wificfg_content_type content_type = HTTP_CONTENT_TYPE_OTHER;
+            bool hostp = false;
+            uint32_t host = IPADDR_NONE;
+
+            for (;;) {
+                int header_length = read_crlf_line(s, buf, sizeof(buf));
+                if (header_length <= 0)
+                    break;
+
+                char *name_end = buf;
+                for (; ; name_end++) {
+                    char c = *name_end;
+                    if (!c || c == ':')
+                        break;
+                }
+                if (*name_end == ':') {
+                    char *value = name_end + 1;
+                    *name_end = 0;
+                    http_header header = intern_http_header(buf);
+                    value = skip_whitespace(value);
+                    switch (header) {
+                    case HTTP_HEADER_HOST:
+                        hostp = true;
+                        host = ipaddr_addr(value);
+                        break;
+                    case HTTP_HEADER_CONTENT_LENGTH:
+                        content_length = strtol(value, NULL, 10);
+                        break;
+                    case HTTP_HEADER_CONTENT_TYPE:
+                        content_type = intern_http_content_type(value);
+                        break;
+                    default:
+                        break;
+                    }
+                }
+            }
+
+            if (hostp && host == IPADDR_NONE) {
+                /* Redirect to an IP address. */
+                wificfg_write_string(s, http_redirect_header_softap);
```

Comment:
@thekr1s deze regel zorgt er voor dat je telefoon de sign in laat zien

Reply:
En ik ben wel benieuwd wat de logging was toen je telefoon raar deed. Ik ga het thuis even proberen te reproduceren